spec/functions/authz_core_config_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'apache::authz_core_config' do
+  let(:input) do
+    {
+      'Require' => [
+        'user foo',
+        'user bar',
+      ]
+    }
+  end
+
+  it { is_expected.to run.with_params(nil).and_raise_error(StandardError) }
+  it { is_expected.to run.with_params([]).and_raise_error(StandardError) }
+  it { is_expected.to run.with_params({}).and_return([]) }
+  it { is_expected.to run.with_params(input).and_return(['Require user foo', 'Require user bar']) }
```

Comment:
Can we add a test with a combination of nested requirements (e.g. the one of the examples) that prove the code works as expected and ensure we don't break the recursion in the future?